network/network_common.go

```diff
@@ -15,9 +15,60 @@ func (self *Network) Name() string {
 
 func (self *Network) Collect() (result interface{}, err error) {
 	result, err = getNetworkInfo()
+	interfaces := getMultiNetworkInfo()
+	if interfaces != nil {
+		interfaceMap, ok := result.(map[string]interface{})
+		if !ok {
+			return
+		}
+		interfaceMap["interfaces"] = interfaces
+	}
 	return
 }
 
+func getMultiNetworkInfo() (multiNetworkInfo []map[string]interface{}) {
+	ifaces, err := net.Interfaces()
+
+	if err != nil {
+		return nil
```

Comment:
Is there a reason that this error is silenced? If there is none, it could be propagated and handled in `Collect()`?

Reply:
that would because the same error would already have been raised by the original code from `getNetworkInfo` that runs first. will try to address this.

Reply:
https://github.com/DataDog/gohai/commit/36a923f33534e9d304b16a0e933b4ce6bebc2092